frodo.js

```diff
@@ -3,82 +3,78 @@
  *
```

Comment:
SCRIPT-Consider extracting this repeated logic into a separate function to avoid duplication.